Mysterious CI Failure - Certain Text in PR's Description Results in Failure to Identify MATLAB Root

In #3358, I encountered the following [mysterious pre-merge CI failure](https://drake-jenkins.csail.mit.edu/job/experimental/4407/compiler=gcc,label=linux_matlab,type=release/) in configuration `gcc_linux_matlab`:

```
CMake Error at /home/ubuntu/workspace/experimental/c5b85b52/cmake/mex.cmake:102 (string):
  string sub-command STRIP requires two arguments.
Call Stack (most recent call first):
  CMakeLists.txt:143 (mex_setup)


SWIG_EXECUTABLE: /home/ubuntu/workspace/experimental/c5b85b52/build/install/bin/swig
Checking if Swig has Matlab support... yes
Found Swig with Matlab support
CMake Error at /home/ubuntu/workspace/experimental/c5b85b52/build/install/share/cmake/SwigMatlab.cmake:43 (message):
  Please run 'mex_setup(REQUIRED)' which is provided by mex.cmake in
  https://github.com/RobotLocomotion/cmake before using this macro.
Call Stack (most recent call first):
  bindings/swig/CMakeLists.txt:55 (add_swig_matlab_module)
```

To debug this, I [instrumented](https://github.com/RobotLocomotion/drake/pull/3358/commits/b767d4ba06e97a3b98a9b3ef27e8e9c4ca428bfd) `mex.cmake` with a debug print statement. This resulted in the following being logged:

```
18:35:00 -- ************************** MATLAB ROOT: _matlab_root=This change was shown to decrease build times by
```

Since `_matlab_root` appeared to get part of the PR's description, I removed the following text from the description:

> Using an HP z640 workstation with an Intel(R) Xeon(R) CPU E5-2620 v3 @ 2.40GHz, 40GB of Buffered DDR4 1866 MHz RAM, and commit 6d1f32d1c351e4c646009c3ee6e3e906bb2a4395, the build time and memory footprint without dReal is ~14 minutes and 5.1GB:
> 
> ```
> $ cd drake-distro
> $ rm -rf build; externals
> $ git reset --hard HEAD
> $ mkdir build; cd build
> $ cmake .. -DCMAKE_BUILD_TYPE:STRING=RelWithDebInfo -DDISABLE_MATLAB=TRUE -DWITH_DREAL=FALSE -DWITH_SNOPT=FALSE 
> $ /usr/bin/time --format=%E,%U,%M make -j12
> 13:55.23,4906.59,5097528
> ```

This resulted in the PR's pre-merge CI tests to pass: https://drake-jenkins.csail.mit.edu/job/experimental/4412/.

Thus, I believe there is a bug in `mex.cmake` that arises when the PR's description contains certain text.

--- a/cmake/mex.cmake
+++ b/cmake/mex.cmake
@@ -79,38 +79,35 @@ function(mex_setup)
   find_program(matlab matlab)
   if ( NOT matlab )
     if (isrequired GREATER -1)
-      message(FATAL_ERROR "matlab is REQUIRED, but I could not find the matlab executable.")
+      message(FATAL_ERROR "Could NOT find MATLAB main program executable")
     endif()
-    message(STATUS "Could not find matlab executable.  mex support will be disabled")
+    message(STATUS "Could NOT find MATLAB main program executable. MEX support will be disabled.")
     return()
   endif()
-  if ( WIN32 )
-    # matlab -n is not supported on windows (asked matlab for a work-around)
-    get_filename_component(_matlab_root ${matlab} PATH)
-    get_filename_component(_matlab_root ${_matlab_root} PATH)
-    find_program(winmat NAMES MATLAB PATHS ${_matlab_root}/bin/win32 ${_matlab_root}/bin/win64 NO_DEFAULT_PATH) # replace bin\matlab.exe with bin\win**\MATLAB.exe
 
-#   todo: consider replacing the line in drake/Makefile with this, but then drake-admin/postProcessCTest would have to run cmake and parse the output
-#    find_program(python NAMES python)
-#    set(python "${python}" CACHE FILEPATH "${python}")
-  else()
-    execute_process(COMMAND ${matlab} -e COMMAND grep -e "MATLAB=" COMMAND cut -d "=" -f2 OUTPUT_VARIABLE _matlab_root TIMEOUT 60)
-  endif()
-  if (NOT _matlab_root)
-    message(FATAL_ERROR "Failed to extract MATLAB_ROOT")
-  endif()
-  string(STRIP ${_matlab_root} MATLAB_ROOT)
+  set(matlab "${matlab}" CACHE FILEPATH "MATLAB main program executable")
 
-  set(matlab "${matlab}" CACHE FILEPATH "${matlab}")
+  # logic from upstream FindMatlab.cmake...
+
+  # resolve symlinks
+  get_filename_component(MATLAB_ROOT "${matlab}" REALPATH)
+
+  # get the directory (the command below has to be run twice)
+  # this will be the matlab root
+  get_filename_component(MATLAB_ROOT "${MATLAB_ROOT}" DIRECTORY)
+  get_filename_component(MATLAB_ROOT "${MATLAB_ROOT}" DIRECTORY) # Matlab should be in bin
+
   find_program(mex NAMES mex mex.bat HINTS ${MATLAB_ROOT}/bin NO_DEFAULT_PATH)
   if (NOT mex)
-     message(FATAL_ERROR "Failed to find mex executable")
+     message(FATAL_ERROR "Could NOT find MEX compiler")
   endif()
 
   find_program(mexext NAMES mexext mexext.bat HINTS ${MATLAB_ROOT}/bin)
-  execute_process(COMMAND ${mexext} OUTPUT_VARIABLE MEX_EXT OUTPUT_STRIP_TRAILING_WHITESPACE)
+  if (mexext)
+    execute_process(COMMAND ${mexext} OUTPUT_VARIABLE MEX_EXT OUTPUT_STRIP_TRAILING_WHITESPACE)
+  endif()
   if (NOT MEX_EXT)
-     message(FATAL_ERROR "Failed to extract MEX_EXT")
+     message(FATAL_ERROR "Could NOT determine binary MEX filename extension")
   endif()
 
   find_file(simulink_FOUND NAMES simstruc.h HINTS ${MATLAB_ROOT}/simulink/include)
@@ -417,6 +414,5 @@ function(compare_compilers outvar compiler1 compiler2)
   endif()
 
 endfunction()
-
 
 include(CMakeParseArguments)

